### PR TITLE
remove deprecated model hooks

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

remove deprecated model hooks

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: docs/source/hooks.rst [hunk 1]
```diff
@@ -40,15 +40,13 @@ Training loop
 ^^^^^^^^^^^^^
 
 - :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_epoch_start`
-- :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_batch_start`
```
Comment (ydcjeff) [inline docs/source/hooks.rst:43]: shall we add `on_train_batch_start`?

File: docs/source/hooks.rst [hunk 2]
```diff
@@ -40,15 +40,13 @@ Training loop
...
 - :meth:`~pytorch_lightning.core.lightning.LightningModule.tbptt_split_batch`
 - :meth:`~pytorch_lightning.core.lightning.LightningModule.training_step`
 - :meth:`~pytorch_lightning.core.lightning.LightningModule.training_step_end` (optional)
 - :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_before_zero_grad`
 - :meth:`~pytorch_lightning.core.hooks.ModelHooks.backward`
 - :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_after_backward`
 - ``optimizer.step()``
-- :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_batch_end`
```
Comment (ydcjeff) [inline docs/source/hooks.rst:51]: here `on_train_batch_end` too?

File: docs/source/hooks.rst (hunk 1 above)
Comment (Borda) [reply docs/source/hooks.rst:43]: Good point, mind make it as suggestion here?

Comment (ydcjeff) [reply docs/source/hooks.rst:43]: Github shows `Applying suggestions on deleted line is not supported.`
```
- :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_train_batch_start`
```

File: docs/source/hooks.rst (hunk 2 above)
Comment (ydcjeff) [reply docs/source/hooks.rst:51]:
```
- :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_train_batch_end`
```

**Comment by williamFalcon** [comment]
@Borda follow up on the next PR with adding those missing hooks
- :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_train_batch_start`	etc

**Pull request merged**